File: modules/springWebFlux2023/src/main/java/eu/lnslr/example2023/RestApplication.java
package eu.lnslr.example2023;

import eu.lunisolar.lava.lang.utils.Collections4U;
import eu.lunisolar.lava.meta.linked_data.SchemaLoader;
import eu.lunisolar.lava.rdf.api.RdfManager;
import eu.lunisolar.lava.rdf.spring.starter.RdfStarterProperties;
import lombok.NonNull;
import lombok.extern.log4j.Log4j2;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.EnableAspectJAutoProxy;
import org.springframework.context.annotation.Import;
import org.springframework.scheduling.annotation.EnableScheduling;

import java.nio.file.Paths;
import java.util.Properties;

@SpringBootApplication
@Import({
        RestConfiguration.class
})
@EnableAspectJAutoProxy
@EnableScheduling
@Log4j2
public class RestApplication {


    public static void main(String... args) {

        SchemaLoader.initializeWithCcl();
        Collections4U.initializeWithCcl();
        RdfManager.initializeWithCcl();

        SpringApplication application = new SpringApplication(RestApplication.class);

        Properties props = new Properties();
        setPropertiesForSpring(props);
        application.setDefaultProperties(props);

        application.run(args);

    }

    private static void setPropertiesForSpring(Properties props) {
        var datasetUrl = getTriplestoreUrl();
        log.info("Triplestore: {}", datasetUrl);
        props.put(RdfStarterProperties.CONNECTION_URL, datasetUrl);
    }

    private static @NonNull String getTriplestoreUrl() {
        return "lava-rdf:jena:memory:/new";
    }


}

File: modules/springWebFlux/src/main/kotlin/eu/lnslr/example/RestApplication.kt
package eu.lnslr.example

import eu.lnslr.example.executor.ExecutorConfiguration
import eu.lnslr.example.tasks.TasksConfiguration
import eu.lunisolar.lava.data.types.xsd.XS
import eu.lunisolar.lava.lang.utils.Collections4U
import eu.lunisolar.lava.meta.linked_data.SchemaLoader
import eu.lunisolar.lava.rdf.api.RdfManager
import eu.lunisolar.lava.rdf.spring.starter.RdfStarterProperties
import eu.lunisolar.magma.func.supp.Be
import eu.lunisolar.magma.func.supp.check.Checks.check
import org.apache.logging.log4j.kotlin.Logging
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.EnableAspectJAutoProxy
import org.springframework.context.annotation.Import
import org.springframework.scheduling.annotation.EnableScheduling
import java.nio.file.Paths
import java.util.*

@SpringBootApplication
@Import(
    RestConfiguration::class,
    TasksConfiguration::class,
    ExecutorConfiguration::class
)
@EnableAspectJAutoProxy
@EnableScheduling
internal class RestApplication : Logging {


}

object log : Logging {

}

fun main(args: Array<String>) {

    SchemaLoader.initializeWithCcl();
    Collections4U.initializeWithCcl()
    RdfManager.initializeWithCcl()

    val path = Paths.get("").toAbsolutePath().toString()
    log.logger.info("Starting in: $path")

    val urlPath = path.replace("\\", "/")

    var dsUrl = "lava-rdf:jena:tdb2:file:///$urlPath/TDB2/"

    log.logger.info("Triplestore: $dsUrl")

    // just to load XS
    check(XS.DURATION.identity()).`must$`(Be::`equal$`, "http://www.w3.org/2001/XMLSchema#duration")

    // lava-rdf:jena:tdb2:file:///D:/p/.r/tech.lnkd.editor.intellij/

    runApplication<RestApplication>(*args) {
        val props = Properties()
        props[RdfStarterProperties.CONNECTION_URL] = dsUrl
        setDefaultProperties(props)
    }
}
